docs/design/fabric-v2+/fpc-lifecycle-v2.puml

```diff
@@ -18,119 +15,126 @@ title Foot Box removed
+    This box abstracts all other orgs and components.
+    Actions are supposed to be performed by channel participants.
+end note
 end box
 
-entity Orderer
-
+entity Orderer order 80
 
-Admin_Org1 -> Admin_Org1 : package
-note left
-	**Chaincode package**
-	as tar.gz with
-	with label mycc-1
-	//ECC_MR: mrenc_1//
+note over Admin
+    Admin uses the fpc-peer wrappers, which abstract the Fabric Peer CLI.
 end note
 
-Admin_Org1 -> Peer1_Org1 : install package
-return packageID: mycc_1:hash
-Admin_Org1 -> Peer2_Org1 : install package
-return packageID
+group chaincode package
+Admin -> FPC_Peer1_CLI_Wrapper : lifecycle chaincode package
 note right
-	chaincode package identifier
-	is mycc-1:hash
+    **Chaincode package**
+    as tar.gz with
+    metadata.json = <path:(enclaves_path), type:(cc_lang), label:(cc_label), sgx_mode: (sim/hw)>
+    code.tar.gz : compressed file with enclave binary and mrenclave
 end note
-
-Admin_Org2 -> Admin_Org2 : package
-note left
-	**Chaincode package**
-	as tar.gz with
-	with label mycc-1
-	//ECC_MR: mrenc_1//
+return
+end /' group chaincode package '/
+
+group install
+Admin -> FPC_Peer1_CLI_Wrapper ++: lifecycle chaincode install
+FPC_Peer1_CLI_Wrapper -> Peer1_CLI ++: lifecycle chaincode install
+return packageId
+note over FPC_Peer1_CLI_Wrapper
+    The following bookking operations help remember (across lifecycle calls)
+    that a package (id) corresponds to a FPC chaincode. Similarly, the
+    approve will later store data for an FPC chaincode installed locally,
+    and the commit checks for data of a locally-approved FPC chaincode.
+    Also, it should be noted that
+    - the install lifecycle op is peer-specific, while
+    - the approveformyorg is org-specific, to be run on at most one org's peer
+    - the and commit op is expected to run on a single peer
+    Hence, the current bookeeping ops expect the approve and commit ops to
+    run on peers where the FPC chaincode has been previously installed.
 end note
-
-Admin_Org2 -> Peer1_Org2 : install package
-return packageID: mycc_1:hash
-Admin_Org2 -> Peer2_Org2 : install package
-return packageID
-
-Admin_Org3 -> Admin_Org3 : package
-note left #FFAAAA
-	**Chaincode package**
-	as tar.gz with
-	with label mycc-1
-	//ECC_MR: mrenc_XXX//
+FPC_Peer1_CLI_Wrapper -> FPC_Peer1_CLI_Wrapper : retrieve cc_language from package
+FPC_Peer1_CLI_Wrapper -> FPC_Peer1_CLI_Wrapper : if cc_language==fpc then store packageId
+return packageId /' Admin -> FPC_Peer1_CLI_Wrapper '/
+end /' group install '/
+
+ref over Admin, FPC_Peer2_CLI_Wrapper, Peer2_CLI
+    install chaincode on Peer2
+end /' ref '/
+
+group local approve
+Admin -> FPC_Peer1_CLI_Wrapper ++: lifecycle chaincode approveformyorg
+FPC_Peer1_CLI_Wrapper -> Peer1_CLI ++: lifecycle chaincode approveformyorg
+note right
+    **Chaincode definition**
+    Name: cc_name, Version: mrenclave,
+    Sequence: sequence_number, Endorsement Policy: (2of3),
+    Plugins: <custom endorsement/validation plugins>
```

Comment:
Actually, we do not really support (user-specified) custom plugins. So maybe better to explicitly specify the ecc validation plugin (which will stay for MVP and presumably even for any later MSP/X509-style attestation handling)  and a note that no custom endorsement and validation plugins ?


Proposal for change:
```suggestion
    Plugins: <fpc ecc validation plugins>  
    [Note: FPC does not support user-defined endorsement or validation plugins]
```

Reply:
03d533192d7c9c2bfcc6c523700d12d9fead3cda